packages/editor/src/editor-ui/welcome-modal/use-welcome-modal.ts

```diff
@@ -0,0 +1,36 @@
+import { fold } from 'fp-ts/lib/Either'
+import { pipe } from 'fp-ts/lib/function'
+import * as t from 'io-ts'
+import { useEffect, useState } from 'react'
+
+const localStorageKey = 'hasUserSeenWelcomeModal'
```

Comment:
In the `PreferenceContext` I tried to start a new pattern that all editor storage keys start with `serlo-editor::`.
Maybe you could use that here as well or maybe even use the `PreferenceContext` here?
It also uses localStorage and has helper functions to get and set the Preference without using the context.

Then we would just have a new preference, something like:
``` ts
welcomeModalShown: {
    storageKey: 'serlo-editor::welcome-modal-shown',
    defaultValue: false,
  },
```

Reply:
I do like the idea of reusing an existing context, but this isn't exactly a preference, no?

Naming pattern is great, thanks for pointing that out!

Reply:
Yeah… not a preference. But behaves like one in many ways.
We could also call it `showWelcomeModalOnStartup` then it's almost a preference (just one that you can't change right now) 😉 
Your call.

Reply:
If it's my call, then I'd leave it like it is. No need for it to live in a context when it's only used by one component, it's not a preference and this hook is simple anyway.